### DISCUSSION TITLE
how to modify the maxtokens of custom segment

### DISCUSSION THREAD

**Discussion by rudadasir** [langgenius/dify#4071] [Help]
dify/api/core/rag/index_processor/index_processor_base.py
dify/api/core/indexing_runner.py

when i modify the files above
if segmentation["max_tokens"] < 50 or segmentation["max_tokens"] > 1000:
                raise ValueError("Custom segment length should be between 50 and 1000.")” 
                
1000-->2000，nothing changes

so,how can i  custom the maxtokens of segments to 2000?

**Comment by algorithmx** [comment]
hey there, I have searched the keyword "segmentation["max_tokens"]" within the entire project and found that the previously hard-coded 1000 limit has been updated to a configurable `INDEXING_MAX_SEGMENTATION_TOKENS_LENGTH`. I am at Dify version `langgenius/dify-api:0.6.10`. You can add/modify this in the docker-compose.yaml file.
